src/builtins/BuiltinTemporal.cpp

```diff
@@ -60,11 +60,48 @@ static Value builtinTemporalPlainDateConstructor(ExecutionState& state, Value th
     if (!newTarget.hasValue()) {
         ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::New_Target_Is_Undefined);
     }
-    Value y = Value(argv[0].asInt32());
-    Value m = Value(argv[1].asInt32());
-    Value d = Value(argv[2].asInt32());
     Value calendar = TemporalCalendar::toTemporalCalendarWithISODefault(state, argc > 4 ? argv[3] : Value());
-    return TemporalPlainDate::createTemporalDate(state, y, m, d, calendar, newTarget);
+    return TemporalPlainDate::createTemporalDate(state, argv[0].asInt32(), argv[1].asInt32(), argv[2].asInt32(), calendar, newTarget);
+}
+
+static Value builtinTemporalPlainDateFrom(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (argc > 1 && !argv[1].isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "options must be object");
+    }
+    Object* optionsObject = argc > 1 ? argv[1].asObject() : new Object(state);
+
+    if (argv[0].isObject() && argv[0].asObject()->isTemporalPlainDateObject()) {
+        TemporalObject::toTemporalOverflow(state, optionsObject);
+        TemporalPlainDate* plainDate = argv[0].asObject()->asTemporalPlainDateObject();
+        return TemporalPlainDate::createTemporalDate(state, plainDate->year(), plainDate->month(), plainDate->day(), plainDate->calendar());
+    }
+    return TemporalPlainDate::toTemporalDate(state, argv[0], optionsObject);
+}
+
+static Value builtinTemporalPlainDateTimeConstructor(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (!newTarget.hasValue()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::New_Target_Is_Undefined);
+    }
+
+    Value calendar = TemporalCalendar::toTemporalCalendarWithISODefault(state, argc > 8 ? argv[9] : Value());
+    return TemporalPlainDateTime::createTemporalDateTime(state, argv[0].asInt32(), argv[1].asInt32(), argv[2].asInt32(), argc > 3 ? argv[3].asInt32() : 0, argc > 4 ? argv[4].asInt32() : 0, argc > 5 ? argv[5].asInt32() : 0, argc > 6 ? argv[6].asInt32() : 0, argc > 7 ? argv[7].asInt32() : 0, argc > 8 ? argv[8].asInt32() : 0, calendar, new Object(state));
```

Comment:
I would prefer an assert for `argc > 2`
Is there some code which ensures the arguments are numbers?

Reply:
It will assert if not int32

Reply:
I mean when this function is called, the caller ensures that the arguments are integers (it checks them in advance). Asserts are not user friendly